### PR TITLE
Initial commit of masterpiece creator disco diffusion app

### PR DESCRIPTION
**Pull request by KogayIrina**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you pin the dependency versions?
- [x] Does this work locally and on cloud?
- [x] Did you add instructions to run the App?
- [ ] Did you write any new necessary tests?

## What does this PR do?

Initial commit of the masterpiece creator disco diffusion lightning app.
Run `lightning run app app.py --cloud` if you want to try it.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by aniketmaurya — CHANGES_REQUESTED** [review]
Great app idea! Can we add the source code in the project rather than the git submodule, please?

**Comment by KogayIrina** [comment]
Hi, @aniketmaurya 
Thank you for the comment. I added the source code to this PR, can you check it, please?

**Comment by KogayIrina** [comment]
By the way, I found that the link to the contributor guideline from the PR template is broken.

image


 @aniketmaurya

File: src/masterpiece-creator-app/app.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        result.save_image_tensor_to_file(file_name)
+        self.drive_1.put(file_name)
+        return result.tensor
+
+    def build_model(self):
+      import discoart
+      return discoart
+
+
+app = L.LightningApp(MasterpieceCreator(cloud_compute=L.CloudCompute("gpu")))
```
Comment (aniketmaurya) [inline src/masterpiece-creator-app/app.py:38]: the app is not running locally. We need to wrap the `MasterpieceCreator` work inside a `LightningFlow` object.

```python
class RootFlow(L.LightningFlow):
  def __init__(self):
     super().__init__()
     self.work = MasterpieceCreator(cloud_compute=L.CloudCompute("gpu"))
  
  def run(self, *args, **kwargs) -> None:
     self.work.run()
  
  def configure_layout(self):
     return {"name": "Art", "content": self.work.url}


app = L.LightningApp(RootFlow())
```

**Review by aniketmaurya — APPROVED** [review]
LGTM

**Pull request merged**